app/models/concerns/course/levels_concern.rb

```diff
@@ -41,4 +41,35 @@ def compute_level_number(experience_points)
     level = compute_level(experience_points)
     level ? level.level_number : 0
   end
+
+  # Returns the current level's experience point (exp) threshold for
+  # for a given number of experience_points.
```

Comment:
technically the levels concern is outside the levels model, so... yah.

but wait a minute, @kxmbrian when we designed the framework, did we say anything about starting from 0?